backend/api/src/test/kotlin/software/shonk/application/service/ShorkServiceTest.kt

```diff
@@ -256,4 +253,38 @@ class ShorkServiceTest {
         shorkService.addProgramToLobby(lobbyId, "playerB", "someNewOtherProgram")
         verify(exactly = 1) { shorkService.resetLobby(lobbyId) }
     }
+
+    @Test
+    fun `test get all lobbies`() {
+        val shorkService = ShorkService(MockShork())
+        var lobbyId = 0L
+        shorkService.addProgramToLobby(lobbyId, "playerA", "someOtherProgram")
+        shorkService.createLobby()
+        shorkService.createLobby()
+        lobbyId += 2
+        shorkService.addProgramToLobby(lobbyId, "playerA", "someProgram")
+        shorkService.addProgramToLobby(lobbyId, "playerB", "someOtherProgram")
+        val result = shorkService.getAllLobbies().allLobbies
+
+        assertEquals(3, result.size)
+        result.forEachIndexed { index, lobby ->
+            when (index) {
+                0 -> {
+                    assertEquals(index, lobby.lobbyId)
+                    assertEquals(listOf("playerA"), lobby.playersJoined)
+                    assertEquals("NOT_STARTED", lobby.gameState)
+                }
+                1 -> {
+                    assertEquals(index, lobby.lobbyId)
+                    assertEquals(listOf("no player submission yet"), lobby.playersJoined)
+                    assertEquals("NOT_STARTED", lobby.gameState)
+                }
+                2 -> {
+                    assertEquals(index, lobby.lobbyId)
+                    assertEquals(listOf("playerA", "playerB"), lobby.playersJoined)
+                    assertEquals("FINISHED", lobby.gameState)
+                }
+            }
+        }
+    }
```

Comment:
Please add another test that is a bit more complex.
It should:
- Create 3 lobbies
- Removes a lobby so that it results in a gap in the lobby IDs (i.e. 0, 1, 3, 4)
- Ensures that the `GET /lobby` endpoint returns all remaining lobbies, even with the gap